inginious/frontend/plugins/upcoming_tasks/templates/coming_tasks.html

```diff
@@ -0,0 +1,134 @@
+        {% endif %}
+    </div>
+    {{ template_helper.call('main_menu', template_helper=template_helper) | safe }}
+{% endblock %}
+
+
+
+{% block content %}
+
+<h2>{{ _("My Upcoming Tasks") }}
+<div class="pull-right">
+        <div class="btn-group btn-group-sm">
+            <button class="btn btn-warning" data-toggle="modal" data-target="#timeModal">
+                <i class="fa fa-exchange"></i> {{_("Switch time planner")}}
+            </button>
+        </div>
+    </div>
+</h2>
+
+<div class="modal fade" id="timeModal" tabindex="-1" role="dialog" aria-labelledby="timeModalLabel" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <form method="post">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="timeModalLabel">{{_("Modify time planner")}}</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <p>{{ _("This allows to display only the tasks whose deadline is in a certain temporal proximity.") }}</p>
+                    <label>{{ _("Number of days:") }}</label>
+                    <select class="form-control" id="new_time_planner" name="time_planner">
+                        {% for timeid in time_planner %}
+                            <option value="{{timeid}}">{{ timeid}}</option>
+                        {% endfor %}
+                    </select>
+                </div>
+                <div class="modal-footer">
+                    <button type="button" class="btn btn-secondary" data-dismiss="modal">{{_("Cancel")}}</button>
+                    <button type="submit" class="btn btn-warning">{{_("Save changes")}}</button>
+                </div>
+            </div>
+        </form>
+    </div>
+</div>
+
+
+<div class="list-group">
+	{%if open_courses %}
+        {% set username = user_manager.session_username() %}
+        {% for courseid, course in open_courses.items() %}
+                <div class="row">
+                    <div class="col-xs-12 col-md-8">
+					{{ course.get_name(user_manager.session_language()) }}
+					{{ my_render.render_upcoming_list(template_helper, course, tasks_data, tasks_list, my_render) | safe }}
+
+
+                    </div>
+                    <div class="col-xs-12 col-md-4">
+                        <i class="pull-right">
+                            {% if course.is_lti() %}
+                                {{ _("LTI course") }} -
+                            {% elif not course.is_open_to_non_staff() %}
+                                {{ _("Hidden course") }} -
+                            {% endif %}
+
+                            {% if username in course.get_admins() or user_manager.user_is_superadmin() %}
```

Comment:
Not sure those are relevant information here.

Reply:
Removed. Indeed this page is mainly useful for students. Showing their status in regard to the course (student / administrator / tutor) is not really relevant.